python/llm/src/bigdl/llm/transformers/modelling_bigdl.py

```diff
@@ -107,42 +109,58 @@ def from_pretrained(cls,
 
         :return: a model instance
         """
-        if native:
-            invalidInputError(dtype.lower() in ['int4', 'int8'],
-                              "Now we only support int4 and int8 as date type for weight")
-            ggml_model_path = pretrained_model_name_or_path
-            return cls.GGML_Model(model_path=ggml_model_path,
-                                  **kwargs)
-        else:
-            return cls.HF_Class.from_pretrained(pretrained_model_name_or_path,
-                                                *args, **kwargs)
+        try:
+            module = importlib.import_module(cls.GGML_Module)
+            class_ = getattr(module, cls.GGML_Model)
+            if native:
+                invalidInputError(dtype.lower() in ['int4', 'int8'],
+                                  "Now we only support int4 and int8 as date type for weight")
+                ggml_model_path = pretrained_model_name_or_path
+                model = class_(model_path=ggml_model_path, **kwargs)
+            else:
+                model = cls.HF_Class.from_pretrained(pretrained_model_name_or_path,
+                                                     *args, **kwargs)
+        except ImportError:
+            invalidInputError(
+                "Could not import bigdl-llm library. "
+                "Please install the bigdl-llm library to "
+                "use this native ggml int4 model: pip install bigdl-llm"
+            )
```

Comment:
Do we actually need this? Since if people calling this function can go into this file, it can be reasonably assumed there's bigdl.llm.models?